rust/core/src/driver_manager.rs

```diff
@@ -609,15 +618,27 @@ fn load_driver_from_registry(
     entrypoint: Option<&[u8]>,
 ) -> Result<DriverInfo> {
     const ADBC_DRIVER_REGISTRY: &str = "SOFTWARE\\ADBC\\Drivers";
-    let drivers_key = root.open(ADBC_DRIVER_REGISTRY)?;
+    let drivers_key = root
+        .open(ADBC_DRIVER_REGISTRY)
+        .and_then(|k| k.open(driver_name.to_str().unwrap_or_default()))
+        .map_err(|e| {
+            Error::with_message_and_status(
+                format!("Failed to open registry key: {e}"),
+                Status::NotFound,
+            )
+        })?;
+
+    let entrypoint_val = match drivers_key.get_string("entrypoint") {
+        Ok(s) => Some(s.into_bytes()),
+        Err(_) => None,
+    };
 
     Ok(DriverInfo {
-        driver_name: drivers_key.get_string("name").unwrap_or_default(),
-        entrypoint: entrypoint
-            .or_else(|| drivers_key.get_string("entrypoint").map(|e| e.into_bytes())),
-        version: drivers_key.get_string("version").unwrap_or_default(),
-        source: drivers_key.get_string("source").unwrap_or_default(),
         lib_path: PathBuf::from(drivers_key.get_string("driver").unwrap_or_default()),
+        entrypoint: match entrypoint_val {
+            Some(e) => Some(e),
+            None => entrypoint.map(|s| s.to_vec()),
+        },
```

Comment:
You could use https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.or_else